packages/gatsby/src/schema/type-definitions.ts

```diff
@@ -26,7 +26,7 @@ export type GatsbyResolver<TSource, TArgs = { [argName: string]: any }> = (
 ) => any
 
 export interface IGatsbyConnection<NodeType> {
-  totalCount: number
+  totalCount: number | (() => Promise<number>)
```

Comment:
So when using LMDB this will always be async and with in-memory it will be sync, right? Shouldn't we break this in v4 so it's always async?

I imagine that on the React side of things, graphql will resolve the promise, right?

Reply:
Yes, `graphql` resolves this promise. But also promises are rather slow so if we can return sync result from the resolver - we'd better do this (maybe not in this case but as a rule of thumb)

Reply:
For graphql it's fine but if the connection is used outside of Gatsby it's rather cumbersome to figure out if it's a promise or not but I think we only use these pieces during graphql resolving, right?

Reply:
Yeah, this is only used for GraphQL resolvers right now. But you are right, I think we can change it to:

```js
totalCount: (() => number | Promise<number>)
```

This will work the same with `await` but graphql will pick the fast sync route when the returned value is number.